frontend/test/__support__/e2e/helpers/e2e-bi-basics-helpers.js

```diff
@@ -6,6 +8,43 @@ export function filter({ mode } = {}) {
   initiateAction("Filter", mode);
 }
 
+export function filterField(fieldName) {
+  return cy.findByLabelText(`filter-field-${fieldName}`);
+}
+
+export function filterFieldPopover(fieldName) {
+  cy.findByLabelText(`filter-field-${fieldName}`).within(() => {
+    cy.findByTestId("select-button").click();
+  });
+  return popover();
+}
+
+function changeOperator(subject, operator) {
+  cy.wrap(subject)
+    .findByTestId("operator-select")
+    .click();
+
+  cy.findByTestId("operator-options")
+    .findAllByText(new RegExp(operator, "i"))
+    .first()
+    .click();
+  return cy.wrap(subject);
+}
+
+function changeValue(subject, newValue, placeholder) {
+  cy.wrap(subject).within(() => {
+    const input = placeholder
+      ? cy.findByPlaceholderText(new RegExp(placeholder, "i"))
+      : cy.get("input").first();
+
+    input.clear().type(newValue);
+  });
+  return cy.wrap(subject);
+}
+
+Cypress.Commands.add("changeOperator", { prevSubject: true }, changeOperator);
+Cypress.Commands.add("changeValue", { prevSubject: true }, changeValue);
```

Comment:
@nemanjaglumac would you have any misgivings about adding these cypress commands? This allows us to make filter tests much more concise e.g.

instead of:
```js
     modal().within(() => {
        cy.findByLabelText("City").within(() => {
          cy.findByText("Contains").click();
        });
      });

      popover().within(() => {
        cy.findByText("Is").click();
      });

      modal().within(() => {
        cy.findByLabelText("City").within(() => {
          cy.findByPlaceholderText("Search by City").type(
            "Indianeown, Indian Valley",
          );
        });
```

we can do
```js
    filterField("City")
        .changeOperator("is")
        .changeValue("Indianeown, Indian Valley");
```

Reply:
Related: https://docs.cypress.io/api/cypress-api/custom-commands#Best-Practices

Reply:
we could just make it a single function which avoids adding a global command for a single use case, but I guess makes it less idiomatic

```js
      filterField("City", {
        operator: 'is',
        value: "Indianeown, Indian Valley",
      });
```

Reply:
👍  We use filters a lot in e2e tests. (i've spent many dozens of hours over the past month re-writing our filter tests again and again) This helper isn't just for testing this particular filter functionality: e.g. when we want to test creating a question from a model, we create a model, add a filter, then save the question.